src/resolvers/Mutation.js

```diff
@@ -90,6 +90,23 @@ const Mutation = {
     };
   },
 
+  async updateCharacter(parent, args, { prisma, request }, info) {
+    const userId = getUserId(request);
+    const { colour } = args.data;
+
+    // Ensure that the colour argument is a proper hex colour code
+    if (colour.length !== 7 || colour.charAt(0) !== '#') {
```

Comment:
This will certainly still admit false positives, for example the string "0=&-!@#". 

There are some interesting methods here for checking if a string represents a hex string: https://www.sitepoint.com/community/t/how-to-check-if-string-is-hexadecimal/162739

I am mostly surprised there isn't a javascript native method to do this, or at least a convenient import.


Anyways, if hex strings are too annoying, it might be easier (and in fact more data efficient) to store an RGB number (255,255,255) instead.

Reply:
True, this was a pretty sloppy implementation, it probably would've been better to use a regex... which upon looking at that link, is exactly what I need.

What makes RGB numbers more data efficient? I would have to store them as strings in the DB still, no?

Reply:
3 short ints can take as little as 3 bytes, a hex string would take 7 chars == 7 bytes. So this would effectively cut color storage size in half.

You can store tuples in a DB as three columns, `color_r`, `color_g`, `color_b` for example. This is a standard technique for storing infinite precision rationals in a database, where you store the numerator and denominator as integers in separate columns; rather than computing the float and storing the imprecise result in a single column.

Reply:
I should also say, comparisons between two tuples of 3 ints would take 3 clock cycles, vs comparing two hex strings would take 7 clock cycles. So you'd expect a speed up in performance as well.

Reply:
But this is all more an intellectual nit, I doubt we need to optimize so much for a single field. I was mostly interested in the rigorous check for formatting. 

That being said, it will be harder to change formats/datatypes later on...once we have actual data.

Reply:
That's very interesting info, but for a hobby project I definitely feel like an optimization to this degree is unnecessary lol. Certainly good knowledge though, thanks for sharing!!

Reply:
I've added proper validation of the hex strings using Regex, pls review